File: Timerlogger.UI/src/app/api/invoice.ts
import { InvoiceModel } from "../models/InvoiceModel";

const BASE_URL = "https://localhost:3001/api";

export async function getAllInvoices(projectId: string) {
    const response = await fetch(
      `${BASE_URL}/Invoice/GetAllInvoices?ProjectId=${projectId}`
    );
    return response.json();
  }

export async function postInvoice(
    projectId: String,
    devName: String,
    devDocNumber: String,
    invoiceDate: Date,
    customerName: String,
    customerDocNumber: String
  ) {
    const invoice = new InvoiceModel("",
        projectId,
        devName,
        devDocNumber,
        "",
        invoiceDate,
        customerName,
        customerDocNumber);

    const requestOptions = {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(invoice),
    };
    const response = await fetch(`${BASE_URL}/Invoice`, requestOptions);
    const data = await response.json();
    return data;
  }

File: ContentManager/ContentManager.Domain/Entities/Invoice.cs
using System;
using System.Collections.Generic;

namespace Timelogger.Entities
{
	public class Invoice
	{
		private string id { get; set; }
		public string Id
		{
			get
			{
				if (this.id == null)
				{
					this.id = Guid.NewGuid().ToString();
				}
				return this.id;
			}
			set => this.id = value;
		}
		public string ProjectId { get; set; }
		public string DevName { get; set; }
		public string DevDocNumber { get; set; }

		private string invoiceNumber { get; set; }
		public string InvoiceNumber
		{
			get
			{
				if (this.invoiceNumber == null)
				{
					this.invoiceNumber = Guid.NewGuid().ToString();
				}
				return this.invoiceNumber;
			}
			set => this.invoiceNumber = value;
		}

		public DateTime InvoiceDate { get; set; }

		public string CustomerName { get; set; }
		public string CustomerDocNumber { get; set; }

	}
}